test/conftest.py

```diff
@@ -101,9 +102,17 @@
 requests_cache.install_cache(urls_expire_after={"huggingface.co": timedelta(hours=1), "*": requests_cache.DO_NOT_CACHE})
 
 
-def fail_at_version(version_major, version_minor):
+def fail_at_version(target_major, target_minor):
     """
     Reminder to remove deprecated features.
+    If you're using this fixture please open an issue in the repo to keep track
+    of the deprecated feature that must be removed. This way will be assured that the
```

Comment:
We should explicitly mention here that the issue should be added to the GitHub milestone of the corresponding `target_major`. For example: https://github.com/deepset-ai/haystack/issues/new?assignees=&labels=&title=Remove%20deprecated%20XYZ&milestone=1.14.0
That way, the release manager can easily lookup all open issues affecting the next release based on that milestone using the overview here: https://github.com/deepset-ai/haystack/milestone/18

Reply:
I left that out because the milestone might not yet exist when that deprecation is added.
That might maybe create confusion if you can't actually do that. 

Though to be fair most of the time we'll be the one marking something as deprecated so we should be able to create a new milestone. 🤔

Reply:
I would still mention the milestone in the comment. It might be that the milestone doesn't exist yet, true. If that's the case people can either create it themselves or ask maintainers to do it. Further, we could already create the milestones for 1.15, 1.16 now or is there any reason to wait? Based on our release schedule we even have the dates.
If the don't mention the milestones here and the issues aren't tagged with it, it's difficult for the release manager to have the issue on their radar before creating the release branch. @ZanSara What do you think?

Reply:
Updated, should be clearer now.